substrafl/remote/register/manage_dependencies.py

```diff
@@ -0,0 +1,206 @@
+"""
+Utility functions to manage dependencies (building wheels, compiling requirement...)
+"""
+import logging
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+from typing import List
+
+from substrafl.dependency import Dependency
+from substrafl.exceptions import InvalidDependenciesError
+from substrafl.exceptions import InvalidUserModuleError
+
+logger = logging.getLogger(__name__)
+
+LOCAL_WHEELS_FOLDER = Path.home() / ".substrafl"
+
+
+def build_user_dependency_wheel(lib_path: Path, operation_dir: Path) -> str:
+    """Build the wheel for user dependencies passed as a local module.
+
+    Args:
+        lib_path (Path): where the module is located.
+        operation_dir (Path): where the wheel needs to be copied.
+
+    Returns:
+        str: the filename of the wheel
+    """
+    # sys.executable takes the current Python interpreter instead of the default one on the computer
+    try:
+        ret = subprocess.run(
+            [
+                sys.executable,
+                "-m",
+                "pip",
+                "wheel",
+                str(lib_path) + "/",
+                "--no-deps",
+            ],
+            cwd=str(operation_dir),
+            check=True,
+            capture_output=True,
+            text=True,
+        )
+    except subprocess.CalledProcessError as e:
+        raise InvalidUserModuleError from e
+    wheel_name = re.findall(r"filename=(\S*)", ret.stdout)[0]
+
+    return wheel_name
+
+
+def copy_local_wheels(path: Path, dependencies: Dependency) -> List[str]:
+    """Copy the local modules given by the user, generating the wheel if necessary.
+
+    Args:
+        path (Path): the path where the local wheels will be copied.
+        dependencies (Dependency): Dependency object from which the list of local installable dependencies
+            (wheels or modules) will be extracted.
+
+    Returns: list of wheel paths relative to `path`
+    """
+    path.mkdir(exist_ok=True)
+    with tempfile.TemporaryDirectory(dir="/tmp") as tmp_dir_name:
+        tmp_dir = Path(tmp_dir_name)
+        dependencies_paths = dependencies.copy_dependencies_local_package(dest_dir=tmp_dir)
+        wheel_paths = []
+        for dependency in dependencies_paths:
+            if dependency.__str__().endswith(".whl"):
+                wheel_paths.append(str(dependency))
+                shutil.copy(tmp_dir / dependency, path)
+            else:
+                wheel_name = build_user_dependency_wheel(
+                    Path(dependency),
+                    operation_dir=tmp_dir,
+                )
```

Comment:
Indeed, I tried to improve the way the code was organised in this section, but it's not perfectly clean yet :) 
If that's okay with you, I would like to postpone any "big" changes to a follow-up PR, so that the feature can be merged now before the next release?

Reply:
Sure